Add keyword-arguments extension for flake8

caldwellst opened the pull request:
Simple addition of [flake8-keyword-arguments](https://pypi.org/project/flake8-keyword-arguments/) for linting as referenced in the [Internal Coding Guidelines doc](https://docs.google.com/document/d/1Ub9zFNqJ-_eV2FDXASJARNhLkWoVGTAa8Y8dHbc53DM/edit?usp=sharing).

Tinkaa commented:
Great find indeed! Would agree with Monica to at least set the minimum to 1, as in that case there is not really gonna be ambiguity

caldwellst commented:
Yeah, agreed about minimum being higher than 0, good idea! Set minimum to 1, but realised there are some instances of 2 arguments functions that don't accept keyword arguments. I added exceptions for them and updated the Google Doc to reflect what we have here. Check what I've done (especially for some of the exceptions as maybe I missed ways to pass keyword arguments?) and merge if okay!

Tinkaa reviewed (approved):
Looks good! 
@turnerm would agree that we are over-versioning.. especially with all the changes we gonna make in the near future
